managers/BaseStatusPoller.js

```diff
@@ -0,0 +1,134 @@
+'use strict';
+
+const _ = require('lodash');
+const assert = require('assert');
+const Promise = require('bluebird');
+const eventmesh = require('../data-access-layer/eventmesh');
+const CONST = require('../common/constants');
+const errors = require('../common/errors');
+const logger = require('../common/logger');
+const config = require('../common/config');
+const NotImplementedBySubclass = errors.NotImplementedBySubclass;
+const Conflict = errors.Conflict;
+
+class BaseStatusPoller {
+  constructor(opts) {
+    assert.ok(opts.resourceGroup, `Property 'resourceGroup' is required to start status poller`);
+    assert.ok(opts.resourceType, `Property 'resourceType' is required to start status poller`);
+    assert.ok(opts.validStateList, `Property 'validStateList' is required to start status poller`);
+    assert.ok(opts.validEventList, `Property 'validEventList' is required to start status poller`);
+    assert.ok(opts.pollInterval, `Property 'pollInterval' is required to start status poller`);
+    this.pollers = [];
+    this.resourceGroup = opts.resourceGroup;
+    this.resourceType = opts.resourceType;
+    this.validStateList = opts.validStateList;
+    this.validEventList = opts.validEventList;
+    this.pollInterval = opts.pollInterval;
+    this.init();
+  }
+
+  getStatus() {
+    throw new NotImplementedBySubclass('getStatus');
+  }
+
+  pollStatus(object, intervalId) {
+    const resourceDetails = eventmesh.apiServerClient.parseResourceDetailsFromSelfLink(object.metadata.selfLink);
+    // If no lockedByPoller annotation then set annotation  with time
+    // Else check timestamp if more than specific time than start polling and change lockedByPoller Ip
+    return eventmesh.apiServerClient.getResource({
+        resourceGroup: resourceDetails.resourceGroup,
+        resourceType: resourceDetails.resourceType,
+        resourceId: object.metadata.name,
+      })
+      .then(resourceBody => {
+        const options = _.get(resourceBody, 'spec.options');
```

Comment:
Can we refactor this into a separate function so the code flow is clear.
I guess the code flow in this function is similar:
```
pollStatus()
      getResourceDeatails() // already present
      getLock()
      getStatus()  // already present
      stopIfDone() 
```

Reply:
Done!